extension/src/debugger/languages/dotnet.ts

```diff
@@ -217,26 +284,35 @@ export function createProjectDebuggerExtension(dotNetService: IDotNetService): R
             // Configure debug session with launch profile settings
             debugConfiguration.cwd = determineWorkingDirectory(projectPath, baseProfile);
             debugConfiguration.args = determineArguments(baseProfile?.commandLineArgs, args);
-            debugConfiguration.env = Object.fromEntries(mergeEnvironmentVariables(baseProfile?.environmentVariables, env));
             debugConfiguration.executablePath = baseProfile?.executablePath;
             debugConfiguration.checkForDevCert = baseProfile?.useSSL;
             debugConfiguration.serverReadyAction = determineServerReadyAction(baseProfile?.launchBrowser, baseProfile?.applicationUrl);
 
-            // Build project if needed
-            if (!isSingleFileAppHost(projectPath)) {
+            if (!isSingleFileApp(projectPath)) {
                 const outputPath = await dotNetService.getDotNetTargetPath(projectPath);
-                if ((!(await doesFileExist(outputPath)) || launchOptions.forceBuild) && await dotNetService.getAndActivateDevKit()) {
+                if ((!(await doesFileExist(outputPath)) || launchOptions.forceBuild)) {
                     await dotNetService.buildDotNetProject(projectPath);
                 }
 
                 debugConfiguration.program = outputPath;
+                debugConfiguration.env = Object.fromEntries(mergeEnvironmentVariables(baseProfile?.environmentVariables, env));
             }
             else {
+                // Single file apps should always be built
+                await dotNetService.buildDotNetProject(projectPath);
                 const runApiOutput = await dotNetService.getDotNetRunApiOutput(projectPath);
-                applyRunApiOutputToDebugConfiguration(runApiOutput, debugConfiguration);
+                const runApiConfig = getRunApiConfigFromOutput(runApiOutput, debugConfiguration);
+                debugConfiguration.program = runApiConfig.executablePath;
+
+                if (runApiConfig.env) {
```

Comment:
(nit) can you just always call

```
debugConfiguration.env = Object.fromEntries(mergeEnvironmentVariables(baseProfile?.environmentVariables, env, runApiConfig.env));
```

even if it is null?

Reply:
Done!